src/main/webapp/javascript/RecipeRendering.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Takes the recipe info of a recipe as a JS object and the div that the recipe
+ * card will be added to. Creates only one recipe card. For the method to work,
+ * recipeInfo is expected to have the following data saved as properties,
+ * regardless of the explicit property name:
+ * a recipe name, recipe tags,  recipe description, and recipe ID to access more
+ * data on the recipe page.
+ */
+createRecipeCard = (divId, recipeInfo) => {
```

Comment:
So there is a lot more that will eventually go into this function to render the recipe card, so I wonder if it would it be possible to write a function to detect and encode html tags to display them as text? That way if a user writes HTML, it won't have any effect because it's being handled before it's displayed.

Reply:
Attempt at editing the code as little as possible while still handling this problem has been included.

Reply:
I think it's awesome that you looked into and learned about XSS, but unfortunately, it's really difficult to guarantee that your own string manipulation will protect from any attack. Here's a stackoverflow that shows some examples that don't involve HTML tags: https://stackoverflow.com/questions/22395316/xss-without-html-tags (also here's a crazy long list: https://owasp.org/www-community/xss-filter-evasion-cheatsheet)

I know that in the limited way you're using the user entered text right now, it may be ok, but as you continue to add to this logic, it's possible that you'll open up vulnerabilities in your code, which will be harder to catch as the logic gets more complicated. You should also consider that other people may need to modify this code in the future, and may not have the same knowledge as you, and may unintentionally misuse it.

In general, with security stuff, you almost never want to implement a defense yourself. There's almost always a vetted solution that's a better guarantee than writing your own logic. So I do think you should use `document.createElement` and avoid using `innerHTML`. I know it's more writing, but it'll be safer, and like Alex said, you'll only have to write it once :)

That said, I do think the work/learning you've done is super valuable; knowing about XSS is super handy when doing web dev, so hopefully you don't feel like you've wasted any time. If you more questions, just let me know, I'm definitely happy to chat about this more.

Reply:
I've updated the code to now use JS DOM manipulation functions which don't implement inner HTML to show user inputted content, while still keeping an approach which allows for easy integration of new elements to the recipe cards.